waku/v2/protocol/waku_noise/noise.nim

```diff
@@ -246,4 +254,143 @@ proc decryptNoisePublicKey*(cs: ChaChaPolyCipherState, noisePublicKey: NoisePubl
+    serializedHandshakeMessageLen +=  serializedPk.len
+    # We add its serialization to the concatenation of all serialized public keys in the handshake message 
+    serializedHandshakeMessage.add serializedPk
+    # If we are processing more than 256 byte, we return an error
+    if serializedHandshakeMessageLen > 256:
+      debug "PayloadV2 malformed: too many public keys contained in the handshake message"
+      raise newException(NoiseMalformedHandshake, "Too many public keys in handshake message")
+
+  # We get the transport message byte length
+  let transportMessageLen = self.transportMessage.len
+
+  # The output payload as in https://rfc.vac.dev/spec/35/. We concatenate all the PayloadV2 as 
+  # payload = ( protocolId || serializedHandshakeMessageLen || serializedHandshakeMessage || transportMessageLen || transportMessage)
+  # We declare it as a byte sequence of length accordingly to the PayloadV2 information read 
+  var payload = newSeqOfCap[byte](1 + # 1 byte for protocol ID             
+                                  1 + # 1 byte for length of serializedHandshakeMessage field
+                                  serializedHandshakeMessageLen + # serializedHandshakeMessageLen bytes for serializedHandshakeMessage
+                                  8 + # 8 bytes for transportMessageLen
+                                  transportMessageLen # transportMessageLen bytes for transportMessage
+                                  )
+  
+  # We concatenate all the data
+  # The protocol ID (1 byte) and handshake message length (1 byte) can be directly casted to byte to allow direct copy to the payload byte sequence
+  payload.add self.protocolId.byte
+  payload.add serializedHandshakeMessageLen.byte
+  payload.add serializedHandshakeMessage
+  # The transport message length is converted from uint64 to bytes in Little-Endian
+  payload.add toBytesLE(transportMessageLen.uint64)
+  payload.add self.transportMessage
+
+  return ok(payload)
+
+
+
+# Deserializes a byte sequence to a PayloadV2 object according to https://rfc.vac.dev/spec/35/.
+proc deserializePayloadV2*(payload: seq[byte]): Result[PayloadV2, cstring] 
+  {.raises: [Defect, NoiseMalformedHandshake, NoisePublicKeyError].} =
+
+  # The output PayloadV2
+  var payload2: PayloadV2
+
+  # i is the read input buffer position index
+  var i: uint64 = 0
+
+  # We start reading the Protocol ID
+  payload2.protocolId = payload[i].uint8
+  i+=1
+
+  # We read the Handshake Message lenght (1 byte)
+  var handshakeMessageLen = payload[i].uint64
+  if handshakeMessageLen > 256:
+    debug "Payload malformed: too many public keys contained in the handshake message"
+    raise newException(NoiseMalformedHandshake, "Too many public keys in handshake message")
+  i+=1
+
+  # We now read for handshakeMessageLen bytes the buffer and we deserialize each (encrypted/unencrypted) public key read
+  var
+    # In handshakeMessage we accumulates the deserialized Noise Public keys read
```

Comment:
```suggestion
    # In handshakeMessage we accumulate the deserialized Noise Public keys read
```

Reply:
Addressed in https://github.com/status-im/nim-waku/commit/430f2ecd6720d5835c430e310ffb06ce27f840c7.